python/src/scipp/plot/profiler.py

```diff
@@ -0,0 +1,340 @@
+from matplotlib.collections import PathCollection
+
+
+def profiler(scipp_obj_dict=None,
+             axes=None,
+             masks=None,
+             filename=None,
+             figsize=None,
+             ax=None,
+             cax=None,
+             aspect=None,
+             cmap=None,
+             log=False,
+             vmin=None,
+             vmax=None,
+             color=None,
+             logx=False,
+             logy=False,
+             logxy=False,
+             resolution=None):
+    """
+    Plot a 2D slice through a N dimensional dataset. For every dimension above
+    2, a slider is created to adjust the position of the slice in that
+    particular dimension.
+    """
+
+    sv = Profiler(scipp_obj_dict=scipp_obj_dict,
+                  axes=axes,
+                  masks=masks,
+                  ax=ax,
+                  cax=cax,
+                  aspect=aspect,
+                  cmap=cmap,
+                  log=log,
+                  vmin=vmin,
+                  vmax=vmax,
+                  color=color,
+                  logx=logx or logxy,
+                  logy=logy or logxy,
+                  resolution=resolution)
+
+    if ax is None:
+        render_plot(figure=sv.fig, widgets=sv.vbox, filename=filename)
+
+    return sv
+
+
+class Profiler(Slicer2d):
+    def __init__(self,
+                 scipp_obj_dict=None,
+                 axes=None,
+                 masks=None,
+                 ax=None,
+                 cax=None,
+                 aspect=None,
+                 cmap=None,
+                 log=None,
+                 vmin=None,
+                 vmax=None,
+                 color=None,
+                 logx=False,
+                 logy=False,
+                 resolution=None):
+
+        # Variables for profile plotting from pick need to be set before
+        # calling the init from Slicer2d as it calls update_axes() and
+        # update_slice(), which require these variables to be defined.
+        self.profile_viewer = None
+        self.profile_key = None
+        self.slice_pos_rectangle = None
+        self.profile_scatter = None
+        self.profile_update_lock = False
+        self.ax_pick = None
+        self.log = log
+
+        super().__init__(scipp_obj_dict=scipp_obj_dict,
+                         axes=axes,
+                         masks=masks,
+                         ax=ax,
+                         cax=cax,
+                         aspect=aspect,
+                         cmap=cmap,
+                         log=log,
+                         vmin=vmin,
+                         vmax=vmax,
+                         color=color,
+                         logx=logx,
+                         logy=logy,
+                         resolution=resolution)
+
+        # Initial checks
+        if self.ndim != 3:
+            raise RuntimeError("Picking on a 2d plot is only supported for 3D "
+                               "data.")
```

Comment:
Any reason for this restriction? Would it "just work" to pick a 2d plot from 4d data?

Reply:
It got more and more messy when allowing for any number of dims.
I can't remember right now what it was exactly, but it was enough for me to think "how much is this going to be used in practise? Is it really worth the effort?"
So i would say we leave it as is for now, unless you have a concrete example where 4d is needed.

Reply:
First exampel that comes to mind is kind of what I showed yesterday: `dims=('tube', 'straw', 'pixel', 'tof')`. Typically, users may want to see a `tof` spectrum, and maybe have slider to select the tube, with a 2d plot of straw and pixel? That is, not what I suggested above, but rather a "hybrid" of the existing slice viewer and the profile plot, where 1 dimension is handled using the profile, but the others using the normal slice-viewer mechanisms?

Reply:
This is what I had started doing right at the beginning, when I was talking about having many special cases with sliders.
After a discussion on a daily call, I thought we had decided not to do this, but maybe i misunderstood?

I do think that it would be useful to have any `dim` as a profile. I think it would be more useful than having a 2d plot show up as a subplot in the case of 4d.

Need to think a bit more about how to do this.